### DISCUSSION TITLE
How to spawn single tsserver for monorepo

### DISCUSSION THREAD

**Discussion by pirey** [LazyVim/LazyVim#810] [Q&A]
I'm working on a mono repo with over a hundred packages inside.
The default behavior is `tsserver` will be spawned each time I open a file on a different package inside the mono repo because each package has its own `package.json` and `tscofnig.json` files.

I found a [thread](https://neovim.discourse.group/t/slow-lsp-on-large-ts-monorepo-project-caching/2668/2) that suggests I can change the root_dir detection to find `.git`.
![image](https://github.com/LazyVim/LazyVim/assets/10344542/9ff6ef45-5a4a-4a4e-889b-ee259f5c85d5)

How can I apply it in LazyVim, without breaking other configurations?

Any help would be appreciated. Thanks.

**Comment by Jomik** [comment]
Try something like 
```lua
return  {
  "neovim/nvim-lspconfig",
  opts = function(_, opts)
    opts.servers.tsserver.root_dir = require("lspconfig.util").root_pattern(".git")
  end,
}
```

**Comment by folke** [comment]
Good point. I actually never thought about this. Shouldn't workspace folders pick this up? Seems like a potential bug or something that's not yet implemented in lspconfig.

I'm not sure if tsseerver will work as it should if you start it from the root and then open files inside the monorepo. Maybe it does. Will run some tests myself.

**Comment by Mat-moran** [comment]
Hi,
any news on that, I tried the solution proposed by @Jomik but it launching me an error
```
Failed to run config for nvim-lspconfig
attempt to index field 'server' (a nil value)
```

maybe is a misconfiguration from my side, I just put the code snippet in file inside /plugins folder like with any other plugin.

**Reply by dpetka2001** [reply]
@Mat-moran It should be `opts.servers` instead of `opts.server`.

**Reply by Jomik** [reply]
Woops. Corrected my comment
